examples/hal/quad/main.rs

```diff
@@ -141,7 +141,7 @@ fn main() {
     println!("Surface format: {:?}", surface_format);
     let swap_config = SwapchainConfig::new()
         .with_color(surface_format);
-    let (mut swap_chain, backbuffer) = device.create_swapchain(&mut surface, swap_config);
+    let (mut swap_chain, backbuffer) = device.create_swapchain(&mut surface, swap_config, None);
```

Comment:
@msiglreith @kvark I know you said to just add a new argument to `create_swapchain`, but are you sure about that? Because now you have these random `None`s sprinkled around the codebase and it's not immediately clear what those are for?

Reply:
Could wrap it in an enum.

`pub enum PreviousSwapchain {
  NoSwapchain,
  Swapchain(&Swapchain)
}
`

or something.

Reply:
@hriundel I'm fine with `None` there at the end. I agree this is not idea, but fine.